examples_tests/02.ComputeShader/main.cpp

```diff
@@ -1,152 +1,225 @@
 #define _NBL_STATIC_LIB_
 #include <nabla.h>
 
-#include "../../source/Nabla/COpenGLDriver.h"
-
+#include "../common/CommonAPI.h"
 
 using namespace nbl;
 using namespace core;
 
 
+const char* src = R"(#version 450
+
+layout (local_size_x = 16, local_size_y = 16) in;
+
+layout(push_constant) uniform pushConstants {
+    layout (offset = 0) uvec2 imgSize;
+} u_pushConstants;
+
+layout (set = 0, binding = 0, rgba8) uniform readonly image2D inImage;
+layout (set = 0, binding = 1, rgba8) uniform image2D outImage;
+
+void main()
+{
+	if (all(lessThan(gl_GlobalInvocationID.xy, u_pushConstants.imgSize)))
+	{
+		vec3 rgb = imageLoad(inImage, ivec2(gl_GlobalInvocationID.xy)).rgb;
+		
+		imageStore(outImage, ivec2(gl_GlobalInvocationID.xy), vec4(1, 1, 0, 1));
+	}
+})";
+
+
+int getQueueFamilyIndex(const core::smart_refctd_ptr<nbl::video::IPhysicalDevice>& gpu, uint32_t requiredQueueFlags)
+{
+	auto props = gpu->getQueueFamilyProperties();
+	int currentIndex = 0;
+	for (const auto& property : props)
+	{
+		if (property.queueFlags & requiredQueueFlags)
+		{
+			return currentIndex;
+		}
+		++currentIndex;
+	}
+	return -1;
+}
+
 int main()
 {
-	// create device with full flexibility over creation parameters
-	// you can add more parameters if desired, check nbl::SIrrlichtCreationParameters
-	nbl::SIrrlichtCreationParameters params;
-	params.Bits = 24; //may have to set to 32bit for some platforms
-	params.ZBufferBits = 24; //we'd like 32bit here
-	params.DriverType = video::EDT_OPENGL; //! Only Well functioning driver, software renderer left for sake of 2D image drawing
-	params.WindowSize = dimension2d<uint32_t>(1280, 720);
-	params.Fullscreen = false;
-	params.Vsync = true; //! If supported by target platform
-	params.Doublebuffer = true;
-	params.Stencilbuffer = false; //! This will not even be a choice soon
-	auto device = createDeviceEx(params);
-
-	if (!device)
-		return 1; // could not create selected driver.
-
-	video::IVideoDriver* driver = device->getVideoDriver();
-
-	io::IFileSystem* filesystem = device->getFileSystem();
-	asset::IAssetManager* am = device->getAssetManager();
-
-	core::smart_refctd_ptr<asset::ICPUDescriptorSetLayout> ds0layout;
+	constexpr uint32_t WIN_W = 1280;
+	constexpr uint32_t WIN_H = 720;
+	constexpr uint32_t SC_IMG_COUNT = 3u;
+
+	auto win = CWindowT::create(WIN_W, WIN_H, system::IWindow::ECF_NONE);
+
+	video::SDebugCallback dbgcb;
+	dbgcb.callback = &CommonAPI::defaultDebugCallback;
+	dbgcb.userData = nullptr;
+	auto gl = video::IAPIConnection::create(video::EAT_OPENGL, 0, "New API Test", &dbgcb);
+	auto surface = gl->createSurface(win.get());
+
+	auto gpus = gl->getPhysicalDevices();
+	assert(!gpus.empty());
+	auto gpu = gpus.begin()[0];
+	int familyIndex = getQueueFamilyIndex(gpu, video::IPhysicalDevice::E_QUEUE_FLAGS::EQF_GRAPHICS_BIT | 
```

Comment:
you dont need to specify enum scope (ie you can omit `E_QUEUE_FLAGS::`). Not a rule in our conventions, just saying